### ISSUE TITLE
Support slightly less strict semantics

### ISSUE THREAD

**Issue by watzon** [j8r/semantic_compare#1]
Some test cases I've noticed don't work:

```crystal
">6"
">6.0"
"> 6.0.0" # notice the space
"> 6.0.0 && < 83.0.0"
```

As of right now it appears that it doesn't handle spaces well, and it isn't capable of determining that "6" is the same as "6.0.0". I know the second issue could also be fixed upstream in `SemanticVersion` itself. It would also be nice to have support for `&&` in addition to `||`.

**Comment by j8r** [comment]
I guess this will be better done by refactoring the library by using a proper expression parser.

**Comment by watzon** [comment]
Yeah probably :sweat_smile: I'm using this for a project and right now I have to make sure to include fully valid semantic versions, otherwise things break.

**Comment by j8r** [comment]
I am curious, what is your project?

**Comment by watzon** [comment]
https://github.com/watzon/browser

**Comment by j8r** [comment]
It think it is not bad to support a strict syntax. Just tell the user that no space is expected, and everything will be consistent. Also I didn't add an example, no need for `&&`, "> 6.0.0 <83.0.0" is valid.

For supporting numbers like `6`, I agree to support it.

**Comment by watzon** [comment]
I'd settle for that haha
